File: src/main/java/com/github/unchama/util/collection/Iterable.kt
package com.github.unchama.util.collection

inline fun <T, R> Iterable<IndexedValue<T>>.mapValues(f: (T) -> R): List<IndexedValue<R>> =
    map { (index, value) -> IndexedValue(index, f(value)) }

fun <T> Iterable<IndexedValue<T>>.toMap(): Map<Int, T> = map { (index, value) -> index to value }.toMap()

File: src/main/java/com/github/unchama/util/collection/Iterable.scala
package com.github.unchama.util.collection

import kotlin.collections.IndexedValue

object CollectionUtils {
    implicit class IndexedValuesOps[T](val receiver: Iterable[IndexedValue[T]]) {
        def mapValues[R](f: T => R): List[IndexedValue[R]] = receiver.map { case (index, value) => IndexedValue(index, f(value)) }

        def toMap(): Map[Int, T] = receiver.map { case (index, value) => index -> value }.toMap
    }
}
